src/ImageManipulation.php

```diff
@@ -831,7 +865,8 @@ function (AssetStore $store, $filename, $hash, $variant) use ($callback) {
                 // Write from another container
                 if ($result instanceof AssetContainer) {
                     try {
-                        return $store->setFromStream($result->getStream(), $filename, $hash, $variant);
+                        $tuple = $store->setFromStream($result->getStream(), $filename, $hash, $variant);
+                        return [$tuple, $result];
```

Comment:
If you return a different type, won't this potentially cause problems for extension that expect the original type?

Reply:
My rationale here is:

1. This is an “internal” closure that we pass to `->manipulate()`, so I don’t think there’s any BC risk with this as it’s impossible for anyone to call the closure directly.
2. The [other approach I came up with](https://github.com/silverstripe/silverstripe-assets/commit/463bb89b95de329368fa5c561ded39d21ed71338#diff-64fe0e1766e08130ce105afde5d1ad67R853) that didn’t change the signature of the closure caused segfaults on PHP 5.6 😞

Reply:
All good. I didn't click when I saw the conditional logic with the `Deprecation::notice` below.